src/shogun/lib/DynamicArray.h

```diff
@@ -13,30 +13,875 @@
 
 #include <shogun/lib/config.h>
 
-#include <shogun/base/SGObject.h>
 #include <shogun/base/DynArray.h>
 #include <shogun/base/Parameter.h>
+#include <shogun/base/SGObject.h>
+#include <vector>
+
+namespace shogun
+{
+	/** @brief Template Dynamic array class that creates an array that can
+	 * be used like a list or an array.
+	 *
+	 * It grows and shrinks dynamically, while elements can be accessed
+	 * via index.  It is performance tuned for simple types like float
+	 * etc. and for hi-level objects only stores pointers, which are not
+	 * automagically SG_REF'd/deleted.
+	 */
+	template <>
```

Comment:
i know that this follows the clang-format (or i guess that's why it's formatted like this) but our `class_list.cpp.py` doesn't know how to handle this, i.e. template definition should be in one line:
`template <> class CDynamicArray<bool> : public CSGObject`